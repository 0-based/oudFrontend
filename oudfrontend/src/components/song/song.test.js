import React from 'React'
import Song from './song'
import Enzyme, {shallow} from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import checkPropTypes from 'check-prop-types'
Enzyme.configure({adapter: new EnzymeAdapter()});
const fullProps = {
    clickedId: '0',
    handleClick:jest.fn(),
    handlePlay: jest.fn(),
    track:{
        "id": "19",
        "name": "إنسان",
        "artists": [
            {
                "id": "5",
                "name": "حمزة نمرة",
                "type": "shaaby",
                "image": "string"
            },
            {
                "id": "2",
                "name": "Sardeena",
                "type": "Shaaby",
                "image": "string"
            }
            ],
            "albumId": "1",
            "type": "Jazz",
            "audioUrl": "www.Facebook.com"
        }
}
const resp = {
    "id": "1",
    "album_type": "Rap",
    "artists": [
      {
        "id": "5",
        "name": "حمزة نمرة",
        "type": "Shaaby",
        "image": "string"
      },
      {
        "id": "1",
        "name": "wegz",
        "type": "Shaaby",
        "image": "string"
      }
    ],
    "genres": [
      "shaaby"
    ],
    "image": "https://i.scdn.co/image/ab67616d0000b273ef029742aa2f7342b3b5daaf",
    "name": "Insan",
    "release_date": "2020-03-07",
    "tracks": {
      "items": [
        {
          "id": "19",
          "name": "إنسان",
          "artists": [
            {
              "id": "5",
              "name": "حمزة نمرة",
              "type": "shaaby",
              "image": "string"
            },
            {
              "id": "2",
              "name": "Sardeena",
              "type": "Shaaby",
              "image": "string"
            }
          ],
          "albumId": "1",
          "type": "Jazz",
          "audioUrl": "www.Facebook.com"
        },
        {
          "id": "20",
          "name": "سلاح التلميذ",
          "artists": [
            {
              "id": "3",
              "name": "ًWegz el wench",
              "type": "Trap",
              "image": "string"
            },
            {
              "id": "4",
              "name": "Amr Diab",
              "type": "farafeery",
              "image": "string"
            }
          ],
          "albumId": "1",
          "type": "Trap",
          "audioUrl": "www.facebook.com"
        },
        {
          "id": "22",
          "name": "سلاح التلميذ",
          "artists": [
            {
              "id": "3",
              "name": "ًWegz el wench",
              "type": "Trap",
              "image": "string"
            },
            {
              "id": "4",
              "name": "Amr Diab",
              "type": "farafeery",
              "image": "string"
            }
          ],
          "albumId": "1",
          "type": "Trap",
          "audioUrl": "www.facebook.com"
        }],
      },
      "type": "string",
      "released": true
    }

const setup = (props={}) =>{
    return shallow(<Song {...props}/>);
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-testid="${val}"]`);
}
describe('song component', ()=>{
    describe('renders correctly with true props', ()=>{
        let component;
        beforeEach (()=>{
            component = setup(fullProps);
        });
        it('renders Song', ()=>{
            const wrapper = findByTestAttr(component, "song")
            expect(wrapper.length).toBe(1)
        })
    })
    describe('testing prop types', ()=>{
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {...fullProps}, 'prop', Song.name);
            expect(result).toBeUndefined();
        })
        //wrong data for clickedId
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {clickedId:0}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {clickedId:false}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {clickedId:{}}, 'prop', Song.name);
            expect(result).toBeDefined();
        })

        //wrong data for handleClick
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handleClick:0}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handleClick:false}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handleClick:{}}, 'prop', Song.name);
            expect(result).toBeDefined();
        })

        //wrong data for handlePlay
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handlePlay:0}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handlePlay:false}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
        it('passing true props', ()=>{
            const result = checkPropTypes(Song.propTypes, {handlePlay:{}}, 'prop', Song.name);
            expect(result).toBeDefined();
        })
    })
    describe('snapshot test', ()=>{
        it('renders correctly', () => {
            const tree = renderer
              .create(<Song {...fullProps}/>)
              .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
})