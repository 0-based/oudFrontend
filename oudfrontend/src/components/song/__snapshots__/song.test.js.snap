// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`song component snapshot test renders correctly 1`] = `
<button
  className="song row"
  data-testId="song"
  id="song"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="songIcon col-1"
  >
    <button
      className="songButton"
      data-testId="playButton"
      onClick={[Function]}
    >
      <img
        alt="play music icon"
        data-testId="playButtonImage"
        height="14"
        src="musicIcon.png"
        width="12"
      />
    </button>
  </div>
  <div
    className="songInfo col-8"
  >
    <span
      className="whiteText"
      data-testId="songName"
    >
      إنسان
    </span>
    <p
      data-testId="aristsNames"
    >
      <span>
        <span>
          <a
            className="playlistAnchor"
            data-testId="artistName"
            href="/artist/5"
            onClick={[Function]}
          >
            حمزة نمرة
          </a>
          <span
            data-testId="comma"
          >
            , 
          </span>
        </span>
        <span>
          <a
            className="playlistAnchor"
            data-testId="artistName"
            href="/artist/2"
            onClick={[Function]}
          >
            Sardeena
          </a>
          <span
            data-testId="comma"
          >
            , 
          </span>
        </span>
      </span>
      <a
        className="playlistAnchor"
        data-testId="albumName"
        href="/albums/1"
        onClick={[Function]}
      >
        
      </a>
    </p>
  </div>
  <div
    className="col-1"
  >
    <div
      className="dropdown"
      data-testId="dropdown"
    >
      <button
        className="songButton"
        data-testId="dropdownButton"
        id="songDropdownButton"
        onClick={[Function]}
      >
        <h3
          className="whiteText"
          style={
            Object {
              "display": "none",
            }
          }
        >
          ...
        </h3>
      </button>
      <div
        className="dropdownContent"
        data-testId="dropdownList"
        id="dropdownContent"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <a
          className="SongDropdownItem"
          data-testId="dropdownItem"
          href=""
          onClick={[Function]}
        >
          Save to your Liked Songs
        </a>
        <a
          className="SongDropdownItem"
          data-testId="dropdownItem"
          href=""
          onClick={[Function]}
        >
          Add to Queue
        </a>
        <a
          className="SongDropdownItem"
          data-testId="dropdownItem"
          href=""
          onClick={[Function]}
        >
          Add to Playlist
        </a>
      </div>
    </div>
  </div>
  <div
    className="col-2"
  >
    <p
      className="whiteText"
      data-testId="songTime"
    >
      3:34
    </p>
  </div>
</button>
`;
